core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -1226,11 +1231,10 @@ class LogTest {
     log = new Log(logDir, config, logStartOffset = 0L, recoveryPoint = numMessages + 1, scheduler = time.scheduler,
       brokerTopicStats = brokerTopicStats, time = time)
     val segArray = log.logSegments.toArray
-    for (i <- 0 until segArray.size - 1) {
+    for (i <- segArray.indices.dropRight(1)) {
```

Comment:
Instead of `dropRight(1)`, you can call `init` (it's like `tail`, but drops the last element instead of first).